packages/common/container-definitions/src/loader.ts

```diff
@@ -275,13 +387,19 @@ export interface IContainer extends IEventProvider<IContainerEvents>, IFluidRout
 
     /**
      * The server provided ID of the client.
-     * Set once this.connectionState === ConnectionState.Connected is true, otherwise undefined
+     *
+     * Set once {@link IContainer.connectionState} is {@link (ConnectionState:namespace).Connected},
+     * otherwise undefined.
+     *
      * @alpha
      */
-    readonly clientId?: string | undefined;
+    readonly clientId?: string;
```

Comment:
Here I'd prefer

```ts
readonly clientId: string | undefined;
```

Since the property always exists on the container, just its value may be undefined.

Reply:
Fair point. Will update.

Reply:
```suggestion
    readonly clientId: string | undefined;
```

Reply:
Actually, since we export this interface, that would be a breaking change. I'll restore the original state for now. We can consider this separately.